index.html

```diff
@@ -227,6 +227,66 @@ <h4>New sink available</h4>
+    <h2><a><code>MediaDevices</code></a> Extensions</h2>
+    <p>This section specifies additions to the <dfn
+    data-cite="!GETUSERMEDIA#dom-mediadevices"><code>
+    MediaDevices</code></dfn> when the Audio Output Devices API is
+    supported.</p>
+    <div>
+      <pre class="idl">
+      partial interface MediaDevices {
+        Promise&lt;MediaDeviceInfo&gt; requestAudioOutputDevice ();
+      };
+      </pre>
+      <section>
+        <h2>Methods</h2>
+        <!-- Update section to use links to promise resolve/reject spec, permission spec... -->
+        <dl data-link-for="MediaDevices" data-dfn-for="MediaDevices"
+        class="method">
+          <dt><dfn><code>requestAudioOutputDevice</code></dfn></dt>
+          <dd>
+            <p>Prompts the user to select a specific audio output device.</p>
+            <p>When the <code>requestAudioOutputDevice</code> method is called,
+            the User Agent MUST run the following steps:</p>
+            <ol>
+              <li><p>If the method call is not <a
+              href="https://html.spec.whatwg.org/multipage/interaction.html#triggered-by-user-activation">
+              triggered by user activation</a>,
+              return a promise rejected with a <code>DOMException</code> object whose
+              <code>name</code> attribute has the value <code>InvalidStateError</code>.
+              </p></li>
+              <li><p>Let <var>p</var> be a new promise.</p></li>
+              <li>
+                <p>Run the followng steps in parallel:</p>
+                <ol>
+                  <li><p>Probe the User Agent for available audio output devices.</p></li>
+                  <li><p>If there is no audio output devices, reject
+                  <var>p</var> with a new <code>DOMException</code> whose <code>name</code> attribute
+                  has the value <code>NotFoundError</code>.</p></li>
+                  <li><p>Prompt the user to choose an audio output device, with a PermissionDescriptor named "speaker".</p></li>
+                  <li><p>If the result of the request is "denied", reject
+                    <var>p</var> with a new <code>DOMException</code> whose <code>name</code> attribute
+                    has the value <code>NotAllowedError</code> and abort these steps.</p></li>
+                  <li><p>Let <var>deviceInfo</var> be a new <code>MediaDeviceInfo</code> object to represent the selected audio output device.</p></li>
+                  <li><p>Update <var>deviceInfo</var> as done by the
+                  <code>enumeratedevices</code> algorithm steps 2.5.4 and 2.5.5.</p></li>
+                  <li><p>Resolve <var>p</var> with <var>deviceInfo</var>.</p></li>
+                </ol>
+              </li>
+              <li><p>Return <var>p</var>.</p></li>
+            </ol>
+            <p>Once a device is exposed after a call to <code>requestAudioOutputDevice</code>, it MUST be listed by
+            <code>enumeratedevices</code>.</p>
```

Comment:
We should discuss this. I would like it to be for the lifetime of the browsing context.
Issue is that setSinkId would not work when reloading the page. Which might be good so that it is not used as an Oracle. Persistency of device ids needs to be discussed

Reply:
If a site can remember I use my bluetooth headset as a mic, but not as a speaker, that seems like a fail.

I think the site should be able to store both deviceIds and use them later.

We may also have to rethink `groupId`, if we can no longer use _enumerateDevices_ to translate `deviceId` to `groupId` and back. Which begs a question: why aren't they the same?

Reply:
I would like to split the persistency discussion in a follow-up discussion.